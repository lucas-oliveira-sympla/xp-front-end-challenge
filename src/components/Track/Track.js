import React, { Component } from 'react';
import {
  number, string, func, bool,
} from 'prop-types';

import TrackStyled from './Track.styles';

const propTypes = {
  id: string.isRequired,
  num: number.isRequired,
  name: string.isRequired,
  duration: string.isRequired,
  handleClick: func,
  isPlaying: bool,
};

const defaultProps = {
  handleClick: () => {},
};

class Track extends Component {
  constructor() {
    super();

    this.clickHandler = this.clickHandler.bind(this);
  }

  clickHandler(event) {
    const { handleClick, id } = this.props;
    handleClick(id);
  }

  render() {
    const { num, name, duration } = this.props;

    return (
      <TrackStyled onClick={this.clickHandler}>
        <span>{num}</span>
        <span>{name}</span>
        <span>{duration}</span>
      </TrackStyled>
    );
  }
}

Track.propTypes = propTypes;
Track.defaultProps = defaultProps;

export default Track;
