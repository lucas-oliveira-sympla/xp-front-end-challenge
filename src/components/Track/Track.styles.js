import styled from 'styled-components';

import breakpoint from 'styled-components-breakpoint';
import colors from '../../theme/colors';

const TrackStyled = styled.li`
  display: flex;
  justify-content: flex;
  width: 100%;
  font-size: 16px;
  padding-bottom: 25px;
  color: ${colors.primary[300]};

  > span {
    font-family: 'Roboto', sans-serif;
  }

  > span:nth-child(1) {
    width: 50px;
    transition: width 0.3s ease-out;

    &::after {
      content: '.';
      margin-right: 15px;

      ${breakpoint('tablet')`
        margin-right: 30px;
      `}
    }
  }
  > span:nth-child(2) {
    flex-grow: 1;
    color: ${colors.primary[700]};
  }}
  }

  &:hover {
    cursor: pointer;

    > span:nth-child(1) {
      width: 65px;
    }
  }
`;

export default TrackStyled;
