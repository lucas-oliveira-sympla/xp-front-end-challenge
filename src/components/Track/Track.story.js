import React from 'react';
import { storiesOf } from '@storybook/react';

import { BackgroundStyled } from '../../App.styles';

import Track from './Track';

const stories = storiesOf('Track', module);
stories
  .add('single', () => (
    <BackgroundStyled>
      <ol>
        <Track id="id" num={1} name="Nome do album" duration="3:31" />
      </ol>
    </BackgroundStyled>
  ))
  .add('listing', () => (
    <BackgroundStyled>
      <ol>
        <Track id="id" num={1} name="Nome do album" duration="3:31" />
        <Track id="id" num={2} name="Nome do album" duration="1:10" />
        <Track id="id" num={3} name="Nome do album" duration="5:21" />
      </ol>
    </BackgroundStyled>
  ));
