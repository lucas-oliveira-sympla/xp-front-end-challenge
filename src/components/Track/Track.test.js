import React from 'react';
import { mount } from 'enzyme';

import Track from './Track';

describe('<Track />', () => {
  let wrapper;

  describe('with default props', () => {
    beforeAll(() => {
      wrapper = mount(<Track id="id" num={1} name="Nome do album" duration="3:31" />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
