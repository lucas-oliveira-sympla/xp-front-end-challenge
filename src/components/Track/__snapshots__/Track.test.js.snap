// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> with default props should match snapshot 1`] = `
<Track
  duration="3:31"
  handleClick={[Function]}
  id="id"
  name="Nome do album"
  num={1}
>
  <Trackstyles__TrackStyled
    onClick={[Function]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Trackstyles__TrackStyled-sc-45spy5-0",
            "isStatic": false,
            "lastClassName": "etTWSl",
            "rules": Array [
              "display:flex;justify-content:flex;width:100%;font-size:16px;padding-bottom:25px;color:",
              "#999",
              ";> span{font-family:'Roboto',sans-serif;}> span:nth-child(1){width:50px;transition:width 0.3s ease-out;&::after{content:'.';margin-right:15px;",
              [Function],
              "}}> span:nth-child(2){flex-grow:1;color:",
              "#FAFAFA",
              ";}}}&:hover{cursor:pointer;> span:nth-child(1){width:65px;}}",
            ],
          },
          "displayName": "Trackstyles__TrackStyled",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Trackstyles__TrackStyled-sc-45spy5-0",
          "target": "li",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <li
        className="Trackstyles__TrackStyled-sc-45spy5-0 etTWSl"
        onClick={[Function]}
      >
        <span>
          1
        </span>
        <span>
          Nome do album
        </span>
        <span>
          3:31
        </span>
      </li>
    </StyledComponent>
  </Trackstyles__TrackStyled>
</Track>
`;
